src/pipeline.ml

```diff
@@ -28,11 +28,31 @@ module Build_unikernel = Build.Make(Packet_unikernel)
 module Build_registry = Build.Make(Docker_registry)
 module Cluster_build = Build.Make(Cluster)
 
-let docker ?(archs=[`Linux_x86_64]) ?(options=Cluster_api.Docker.Spec.defaults) ~sched dockerfile targets =
+type deployment = {
+  branch : string;
+  target : string;
+  services : Cluster.service list;
+}
+
+let make_deployment branch target services = { branch; target; services; }
```

Comment:
Since these arguments have the same name, I'd be inclined to use labels here to ensure the meaning of each argument can be read from the function's type.

```suggestion
let make_deployment ~branch ~target services = { branch; target; services; }
```

An alternative that would accomplish the same thing, imo, would be to just use record literals to enter the config data rather than this helper function.